dali/pipeline/operator/operator.h

```diff
@@ -395,6 +395,34 @@ DALI_DECLARE_OPTYPE_REGISTRY(MixedOperator, OperatorBase);
 
 DLL_PUBLIC std::unique_ptr<OperatorBase> InstantiateOperator(const OpSpec &spec);
 
+namespace detail {
+
+template <typename Backend>
+class OpImplBase {
```

Comment:
I'd prefer to have it here once, rather than having it repeated in every operator. It's not like I am saying every operator should use it, but it's a `detail` utility that comes in handy for `pimpl ` implementations. Open to discussing a better place for it

Reply:
I agree with @szalpal, that this can be muddying the waters with respect to Operator API. Maybe it would be better to suggest that kind of change in separate PR with proper docs and guideline how to use it with other operators?

Reply:
What I'm afraid of is that we'll start using it, just "because it's there". In my opinion, the pimpl in operators implementation, in this form, makes the implementation even more convoluted. We have templated backend there and this interface is not really simple. Probably, if we'd put more thinking into this interface, it could work fine, but IMHO not in this shape. I, personally, would leave this in multiple places as it is, but I understand, that this would make this PR yet harder to merge (That's also why I'm not fan of big PRs ;) ), therefore I'm not pushing it. Just wanted to express my concern, that this is probably bad decision ;)

Reply:
At the very least it deserves some comment on what the purpose of this class is - and no, it's not for implementing operators or even less "for pimpl" - pImpl is a way to remove implementation details from an exported or otherwise public interface - this class is "for providing argument- or type-specific implementation of operators". I would consider any use beyond that to actually be an abuse.
As for the placement - I think that it could be moved to a separate header file. Then it could get some extra utilities that would remove some more boilerplate code (whatever that may be).